pkg/reconciler/taskrun/controller.go

```diff
@@ -66,6 +67,7 @@ func NewController(images pipeline.Images) func(context.Context, configmap.Watch
 			ConfigMapWatcher:  cmw,
 			ResyncPeriod:      resyncPeriod,
 			Logger:            logger,
+			Recorder:          controller.GetEventRecorder(ctx),
```

Comment:
is this just for testing? that's a bit unfortunate if so :( probably worth it tho?

Reply:
Yes, injecting fakes in the context is the same mechanism that we use to inject fake k8s client-go and other fakes.